DOL.WHD.Section14c.Web/src/modules/components/attachmentField/attachmentFieldController.test.js

```diff
@@ -2,17 +2,79 @@ describe('attachmentFieldController', function() {
 
     beforeEach(module('14c'));
 
-    beforeEach(inject(function ($rootScope, $controller) {
+    beforeEach(inject(function ($rootScope, $controller, _$q_, apiService) {
         scope = $rootScope.$new();
+        $q = _$q_;
+        mockApiService = apiService;
 
         attachmentFieldController = function() {
             return $controller('attachmentFieldController', {
-                '$scope': scope
+                '$scope': scope,
+                'apiService': mockApiService
             });
         };
+
+        uploadAttachment = $q.defer();
+        spyOn(mockApiService, 'uploadAttachment').and.returnValue(uploadAttachment.promise);
+
+        deleteAttachment = $q.defer();
+        spyOn(mockApiService, 'deleteAttachment').and.returnValue(deleteAttachment.promise);
+
+
     }));
 
     it('invoke controller', function() {
         var controller = attachmentFieldController();
     });
+
+    it('attachment selected no files', function() {
+        var controller = attachmentFieldController();
+        fileInput = { files: [] };
+        controller.onAttachmentSelected(fileInput);
```

Comment:
This is to test the if branch when no file is provided.  Added asserion that the input should stay the same (no-op). https://github.com/AppliedIS/dol-whd-14c/commit/56fe4dc4ef371d10173fb4237ebcfc0772b8f4dc